File: LC203.cpp
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode() : val(0), next(nullptr) {}
 *     ListNode(int x) : val(x), next(nullptr) {}
 *     ListNode(int x, ListNode *next) : val(x), next(next) {}
 * };
 */
class Solution {
public:
    ListNode* removeElements(ListNode* head, int val) {
        

        if(head == NULL){
            return head;
        }

        if(head->next == NULL){
            if(head->val == val)
                return NULL;
            return head;
        }

        while(head != NULL && head->val == val){
          head = head->next;
        }

        ListNode *curr = head;

        while(curr != NULL && curr->next != NULL){
            
            if(curr->next->val == val){
              ListNode *p = curr->next;
              curr->next = p->next;
              p->next = NULL;
              delete p;
            }
            else
                curr = curr->next;
        }

        return head;
    }
};

File: LC203.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def removeElements(self, head: Optional[ListNode], val: int) -> Optional[ListNode]:

        if head == None:
            return head
        
        if head.next == None:
            if head.val == val:
                return None
            else:
                return head
        
        while head is not None and head.val == val:
            head = head.next
        
        curr = head

        while curr and curr.next:
            if curr.next.val == val:
                p = curr.next
                curr.next = p.next
                p.next = None
                del p
            else:
                curr = curr.next
        

        return head
